pkgs/development/libraries/tpm2-tss/default.nix

```diff
@@ -11,6 +11,20 @@ let
   # might not support the withSystemd option.
   procpsWithoutSystemd = procps.override { withSystemd = false; };
   procps_pkg = if stdenv.isLinux then procpsWithoutSystemd else procps;
+
+  # Configure script expects tools from shadow but we are not actually using
+  # them. Instead of patching the build scripts, just add fake executables
+  # to native build inputs. These would fail the build if called from both POSIX
+  # shell and via native exec system call (albeit for different reasons).
+  # https://github.com/tpm2-software/tpm2-tss/blob/6c46325b466f35d40c2ed1043bfdfcfb8a367a34/configure.ac#L666-L675
+  # https://github.com/tpm2-software/tpm2-tss/blob/6c46325b466f35d40c2ed1043bfdfcfb8a367a34/Makefile.am#L880-L898
+  fakeShadow = runCommand "fake-shadow" { } ''
```

Comment:
Looks to me like it will just [fail with a warning](https://github.com/tpm2-software/tpm2-tss/blob/6c46325b466f35d40c2ed1043bfdfcfb8a367a34/Makefile.am#L988), which we should be able to safely ignore. So we should just be able to remove shadow altogether. Maybe also patch the warning out to be explicit and avoid the noise.

Or maybe even better, switch to sysusers.d.

Reply:
I don’t think `tpm2-tss` package runs `install-dirs` so we don’t even reach this warning. However, configure script fails if we don’t have shadow in PATH. Perhaps we can patch that instead and run autoreconfHook? 🤔
https://github.com/tpm2-software/tpm2-tss/blob/6c46325b466f35d40c2ed1043bfdfcfb8a367a34/configure.ac#L667-L675

Reply:
I don't think removing these dependencies is a problem at all. Nix manages these users. If anything, this should be able to be disabled upstream.

Reply:
Easier to fake:

https://github.com/tpm2-software/tpm2-tss/blob/6c46325b466f35d40c2ed1043bfdfcfb8a367a34/configure.ac#L661

Just create a noop `systemd-sysusers` script on Linux that does nothing, and it'll use that instead of shadow.

Reply:
Actually, we do indeed run `install-dirs` and get the warning because of the Nix sandbox.
```console
$ nix log nixpkgs/nixos-24.05#tpm2-tss | fgrep -C3 'Failed to create the tss user and group'
call make_tss_user_and_group
groupadd: Permission denied.
groupadd: cannot lock /etc/group; try again later.
WARNING Failed to create the tss user and group
(call make_fapi_dirs) && (call set_fapi_permissions)
chown: invalid user: 'tss:tss'
WARNING Failed to create the FAPI directories with the correct permissions
```

I’ve update the PR to patch configure.ac so we print a warning instead of aborting configure phase 😅
I’m not sure why I didn’t do that from the beginning.